solutions.js

```diff
@@ -0,0 +1,52 @@
+//***CHALLENGES***
+
+// Write a function called each that takes in an array and a callback function. 
+var each = function (arr, callback){
+// each should iterate through all items in the array
+	for (var i = 0; i < each.length; i++){
```

Comment:
This is really close. All your logic is correct, but you've just mixed up some variable names. Is `each` a variable we have? Look at what the parameters are called. Remember we are looping through all the items in the `arr` we passed in.